### ISSUE TITLE
Write to .CSV/Excel/Sheets instead of .txt

### ISSUE THREAD

**Issue by xuveki** [xuveki/laps#6]

**Comment by Aspireve** [comment]
Hey Hi @xuveki, could I try working on this issue?
For this we just need to create a csv and keep updating it right?

**Comment by xuveki** [comment]
Hey @Aspireve, sure thing! Yes, creating and updating the file, just like the current "laps-history.txt." The info I was planning on writing: 

- Date program was opened
- Starting time and day
- Ending time and day
- Tasks
- Task times
- Repeated tasks
- Repeated task time totals
- Total time
- Comments.

A design problem with writing to a spreadsheet is figuring out how to present the tasks and times. There's always going to be one total time, one start date, one comment field, etc., but the number of tasks is variable.

**Comment by Aspireve** [comment]
Hey Hi @xuveki 
I actually ran the program but didn't get any txt file in my folder.
So do you mean something like 
Start User Session
- Complete Task One (time, data, stuff)
- Complete task Two (time, data, stuff)
-
-
End user session
Also If the csv file has to be made after the session is complete we could use a pandas data frame

**Comment by Aspireve** [comment]
The dataset could be
SessionData.csv
Session Id, Session Name, Start Time, Comments, Total Time.....

TaskData.csv
Session Id, Task Name, Task Start Time, Task End Time, Repeated times....

How does this type of a dataset sound, it'll just 2 csv files with Session Id as the Primary Key

**Comment by xuveki** [comment]
@Aspireve, Thanks for letting me know about the .txt file not working. Are you on Windows/using the .py file to run the program? I tried that and had trouble too so I fixed some path issues in the create_file_and_write() function. It should now write/create a .txt file using either the exec or .py. Let me know if you still run into any issues with that.

Your breakdown is pretty much it. It would be similar to the output in the .txt file. Example pic below.

<img width="281" alt="image" src="https://user-images.githubusercontent.com/82191661/233868238-3258fd6b-fa2f-4792-80d4-d27e7b282937.png">

I'm not familiar with pandas! I'll check it out soon. 

As for the layout, just the task length is fine. And 2 CSVs could work but it might be annoying to have to switch between 2 sheets in excel to see the data. Something like this could be a good solution. I haven't looked into python to excel so I have no idea how hard this is.

<img width="1055" alt="image" src="https://user-images.githubusercontent.com/82191661/233869818-511e7f70-63c4-4f0f-8c2e-b4a6bd6adab5.png">

**Comment by Aspireve** [comment]
Hey Hi @xuveki!!

Sorry was a bit inactive (Have my college exams going on currently lol)

Actually the layout you mentioned might not be possible to create in a CSV file cause a csv file is basically a file with a rows (as text) and column values as the values separated by a comma (,) and the only operations on the file would be the basic CRUD (Create, read, Update, Delete) actions.
something like this

| (in notepad)  | (in excel)  | 
|:--:|:--:|
| ![image](https://user-images.githubusercontent.com/93852415/234350575-1d41003b-3ebc-4b68-8b44-e299d08fc710.png) | ![image](https://user-images.githubusercontent.com/93852415/234350942-c54e63dc-0633-43f3-9243-f3d603396668.png) |

so making the format you spoke about might not be possible

Instead I would suggest we could follow the data storing in the above format (which I mentioned) and when the user requests to view the data it would create an excel file in this format (the one you told about)

**Comment by xuveki** [comment]
@Aspireve,

No worries! My schedule is hectic right now too, so I can't spend as much time as I'd like on this.

Understood about keeping the file handling to basic CRUD. Any reason for picking pandas over the csv library? I'm not familiar with it, but it seems like people recommend it for more complex work.

With the csv library route, I think something simple like this could work using csv and joining two lists. What do you think?

<img width="653" alt="image" src="https://user-images.githubusercontent.com/82191661/234441341-e908dafb-04d6-4aca-8b0b-fb433b89d6f2.png">

Using a headerList + taskNameList and a valuesList + taskTimeList. What do you think? The writing/appending could just be 
`convertStrings = csv.writer(file_object)`
`convertStrings.writerow(fullHeaderRow)`
`convertStrings.writerow(fullValueRow)`

"Repeated" tasks could be added as well, like in the txt file. It's a dictionary so there might be a different way to implement that.

**Comment by Aspireve** [comment]
Yup exactly we could do this. but kinda having a variable number tasks and time for them might create a bit of a problem
Kinda had studied about this concept called Data-Base Normalization wherein having a variable number of columns might create a problem as the project scales in the future
But still though we could try doing your way aswell cause I don't think having a  variable column length might cause a problem for this project cause we wont be calling for complex queries from the csv file

**Comment by xuveki** [comment]
@Aspireve 

Interesting! Thanks, I didn't think about that being an issue in the future. If you'd like to try a solution that might be more scalable, then we could go for that as well. Either way, I'm game!
